Examples/Statistics/BayesianClassifier.cxx

```diff
@@ -67,15 +67,21 @@ main(int argc, char * argv[])
     return EXIT_FAILURE;
   }
 
-  // input parameters
-  const char * membershipImageFileName = argv[1];
-  const char * labelMapImageFileName = argv[2];
 
   // setup reader
   constexpr unsigned int Dimension = 2;
   using InputPixelType = float;
   using InputImageType = itk::VectorImage<InputPixelType, Dimension>;
-  const auto input = itk::ReadImage<InputImageType>(membershipImageFileName);
+  InputImageType::Pointer input;
+  try
+  {
+    input = itk::ReadImage<InputImageType>(argv[1]);
+  }
```

Comment:
In general it appears preferable to "try-catch" on a higher level, rather than adding a "try-catch" to each statement that might throw an exception.  So in this case, I would rather have the `try` to the begin of the `main` function, and the `catch` at the end, just before `return EXIT_SUCCESS`. 

See also CppCoreGuidelines [Don’t try to catch every exception in every function](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#e17-dont-try-to-catch-every-exception-in-every-function)

Reply:
I completely agree with this. I updated accordingly the 4 files touched here.